x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/syncjob/ConnectorSyncJobIndexService.java

```diff
@@ -87,15 +88,29 @@ public void createConnectorSyncJob(
         PostConnectorSyncJobAction.Request request,
         ActionListener<PostConnectorSyncJobAction.Response> listener
     ) {
+        String connectorId = request.getId();
         try {
-            getSyncJobConnectorInfo(request.getId(), listener.delegateFailure((l, connector) -> {
+            getSyncJobConnectorInfo(connectorId, listener.delegateFailure((l, connector) -> {
                 Instant now = Instant.now();
                 ConnectorSyncJobType jobType = Objects.requireNonNullElse(request.getJobType(), ConnectorSyncJob.DEFAULT_JOB_TYPE);
                 ConnectorSyncJobTriggerMethod triggerMethod = Objects.requireNonNullElse(
                     request.getTriggerMethod(),
                     ConnectorSyncJob.DEFAULT_TRIGGER_METHOD
                 );
 
+                if (Strings.isNullOrEmpty(connector.getIndexName())) {
+                    l.onFailure(
+                        new ElasticsearchStatusException(
+                            "Cannot start a sync for connector ["
+                                + connectorId
+                                + "] with no index attached. Set the [index_name] property for the connector "
+                                + "to enable syncing data.",
+                            RestStatus.BAD_REQUEST
+                        )
+                    );
+                    return;
+                }
```

Comment:
nit: This could be moved to the top of this block so it's instantly obvious what's a failure case